orttraining/orttraining/training_ops/cuda/tensor/gather_grad.cc

```diff
@@ -76,13 +76,10 @@ Status DispatchToGatherGradImpl(
   if (utils::IsPrimitiveDataType<float>(t_data_type)) {
     return DispatchToGatherGradImplByTin<float>(
         tin_data_type, cuda_kernel, num_weights, stride, num_inputs, param_itrs, grad, indices, output);
-  }
-#if !defined(__CUDA_ARCH__) || __CUDA_ARCH__ >= 700
-  else if (utils::IsPrimitiveDataType<MLFloat16>(t_data_type)) {
+  } else if (utils::IsPrimitiveDataType<MLFloat16>(t_data_type)) {
```

Comment:
is this ok to simply remove the check?

Reply:
I think tis' okay. If the GPU is not supposed to support MLFloat16. it will fail anyway, I checked Gemm, Matmul, they don't check the hardware when calling into MLFloat16 code path, am I missing something?

Reply:
I wonder why it 's there initially.

Reply:
@edgchen1 - can you please review this part? Looks like you refactored this code a while ago.

Reply:
i think so too. `__CUDA_ARCH__` is not supposed to be used in host code.
https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html#virtual-architecture-identification-macro
i probably copied it over from somewhere :)

Reply:
should we replace the deleted check with NeedSkipIfCudaArchLowerThan() and/or HasCudaEnvironment() ?

Reply:
I think the condition was always true before. The half versions of GatherGradImpl() are instantiated without any such check in gather_grad_impl.cu. So I think we don't need a check here.